Game/utils/crashlog.cpp

```diff
@@ -92,25 +97,78 @@ void CrashLog::dumpStack(const char *sig) {
 #ifdef __WINDOWS__
   dbg::symdb          db;
   dbg::call_stack<64> traceback;
-
+#endif
   if(sig==nullptr)
     sig = "SIGSEGV";
 
   std::cout.setf(std::ios::unitbuf);
   std::cout << std::endl << "---crashlog(" <<  sig   << ")---" << std::endl;
   writeSysInfo(std::cout);
+#ifdef __WINDOWS__
   traceback.collect(0);
   traceback.log(db, std::cout);
+#elif __LINUX__
+  tracebackLinux(std::cout);
+#endif
   std::cout << std::endl;
 
   std::ofstream fout("crash.log");
   fout.setf(std::ios::unitbuf);
   fout << "---crashlog(" <<  sig << ")---" << std::endl;
   writeSysInfo(fout);
+#ifdef __WINDOWS__
   traceback.log(db, fout);
+#elif __LINUX__
+  tracebackLinux(fout);
 #endif
   }
 
+std::string CrashLog::demangle(const void* frame, const char* symbol) {
```

Comment:
Using std::string in crash-log is risky: string do heap-allocation, while heap can be already corrupted. 
May be just emplace this code into `void CrashLog::tracebackLinux`

Reply:
I agree, the reworked approach should prevent these allocations.